modules/apiserversource-yaml.adoc

```diff
@@ -0,0 +1,217 @@
+
+.Procedure
+
+. Create a service account, cluster role and cluster binding for the ApiServerSource.
++
+You can do this by creating a file named `authentication.yaml` and copying the following sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default <1>
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default <1>
+----
++
+<1> Adjust this namespace to the namespace that you have selected for installing ApiServerSource.
++
+[NOTE]
+====
+If you want to re-use an existing service account with the appropriate permissions, you must modify the `authentication.yaml` for that service account.
+====
++
+After you have created the `authentication.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename authentication.yaml
+----
+
+. Create an ApiServerSource event source.
++
+You can do this by creating a file named `k8s-events.yaml` and copying the following sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1beta1
+      kind: Broker
+      name: default
+----
++
+After you have created the `k8s-events.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename k8s-events.yaml
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log.
++
+Copy the following sample YAML into a file named `service.yaml`:
++
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+  namespace: default
+spec:
+  template:
+    spec:
+      containers:
+        - image: quay.io/openshift-knative/knative-eventing-sources-event-display
+----
++
+After you have created the `service.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename service.yaml
+----
+
+. Create a trigger from the `default` broker to filter events to the service created in the previous step.
++
+You can create the trigger by creating a file named `trigger.yaml` and copying the following sample code into it:
++
+----
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: event-display-trigger
+  namespace: default
+spec:
```

Comment:
do we want to cover the `annotation` for broker creation ? 

```
  annotations:
    knative-eventing-injection: enabled
```

Reply:
I actually think we should keep it separate since we mention creating the broker as a prerequisite for the ApiServerSource, and this will just complicate changes further once we add things like MT broker in 0.14.